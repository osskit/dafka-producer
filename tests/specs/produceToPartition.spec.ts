import delay from 'delay';

import type {Orchestrator} from '../testcontainers/orchestrator.js';
import {start} from '../testcontainers/orchestrator.js';
import {consume} from '../services/consume.js';

const topic = 'my-topic';

describe('tests', () => {
    let orchestrator: Orchestrator;

    beforeEach(async () => {
        orchestrator = await start(
            {
                KAFKA_BROKER: 'kafka:9092',
                MAX_BLOCK_MS: '1000',
            },
            [topic]
        );
    }, 5 * 60 * 1000);

    afterEach(async () => {
        if (!orchestrator) {
            return;
        }
        await orchestrator.stop();
    });

    it('produce to partition', async () => {
        orchestrator.dafkaProducer.produce([
            {
                topic,
                partition: 2,
                key: 'thekey',
                value: {data: 'foo'},
            },
        ]);

        await delay(5000);

        await expect(consume(orchestrator.kafkaClient, topic)).resolves.toMatchSnapshot();
    });
});
