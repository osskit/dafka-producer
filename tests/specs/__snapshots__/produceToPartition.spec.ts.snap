// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests produce to partition 1`] = `
{
  "headers": {},
  "partition": 2,
  "value": {
    "data": "foo",
  },
}
`;
